package/network/utils/comgt/files/ncm.sh

```diff
@@ -117,21 +127,23 @@ proto_ncm_setup() {
 	}
 
 	echo "Starting network $interface"
-	json_get_vars connect
-	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
-		echo "Failed to connect"
-		proto_notify_error "$interface" CONNECT_FAILED
-		return 1
-	}
-
 	echo "Setting up $ifname"
-	
+
 	proto_init_update "$ifname" 1
 	proto_add_data
 	json_add_string "manufacturer" "$manufacturer"
 	proto_close_data
 	proto_send_update "$interface"
 
+	json_load "$(cat /etc/gcom/ncm.json)"
+	json_select "$manufacturer"
+	json_get_vars connect
+	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
+		echo "Failed to connect"
+		proto_notify_error "$interface" CONNECT_FAILED
+		return 1
+	}
+
```

Comment:
Hi Matti,
here the log sending the commands manually:
```
Terminal ready
AT+CFUN=1
OK
AT+CPIN="****"
OK
AT+CGDCONT=1,"IP","web.vodafone.de",0,0
OK
AT+CGACT=1,1
OK
AT+CGDATA="M-MBIM",1,1
FATAL: term closed
```
and here the dmsg:
```
[   24.064006] device eth0.1 entered promiscuous mode
[   24.074022] device eth0 entered promiscuous mode
[   24.114618] br-lan: port 1(eth0.1) entered forwarding state
[   24.125964] br-lan: port 1(eth0.1) entered forwarding state
[   26.117111] br-lan: port 1(eth0.1) entered forwarding state
[   29.660784] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
[   29.707597] device wlan0 entered promiscuous mode
[   30.567426] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
[   30.580608] br-lan: port 2(wlan0) entered forwarding state
[   30.591865] br-lan: port 2(wlan0) entered forwarding state
[   32.587030] br-lan: port 2(wlan0) entered forwarding state
[   67.747101] random: nonblocking pool is initialized
root@lede:~# dmesg |tail
[  612.377046] usb 1-1: new high-speed USB device number 3 using ehci-platform
[  612.559495] cdc_ether 1-1:1.0 usb0: register 'cdc_ether' at usb-101c0000.ehci-1, CDC Ethernet Device, 02:00:ff:aa:aa:aa
[  612.604199] option 1-1:1.2: GSM modem (1-port) converter detected
[  612.617469] usb 1-1: GSM modem (1-port) converter now attached to ttyUSB0
[  612.667960] option 1-1:1.3: GSM modem (1-port) converter detected
[  612.681000] usb 1-1: GSM modem (1-port) converter now attached to ttyUSB1
[  612.728848] option 1-1:1.4: GSM modem (1-port) converter detected
[  612.741946] usb 1-1: GSM modem (1-port) converter now attached to ttyUSB2
[  612.798695] option 1-1:1.5: GSM modem (1-port) converter detected
[  612.811776] usb 1-1: GSM modem (1-port) converter now attached to ttyUSB3
root@lede:~#
```
My impression is that if I send the ``AT+CGDATA="M-MBIM",1,1`` without having the net IF setup something go wrong and the cdc_ether driver is reset.